packages/core/vue/src/useWallet.ts

```diff
@@ -149,34 +137,33 @@ export const createWalletStore = ({
     // Select a wallet by name.
     const select = async (walletName: WalletName): Promise<void> => {
         if (name.value === walletName) return;
-        if (adapter.value) await adapter.value.disconnect();
         name.value = walletName;
     };
 
-    // Handle the adapter events.
+    // Handle the wallet events.
     const onDisconnect = () => (name.value = null);
     const onConnect = () => {
-        if (!adapter.value) return;
-        publicKey.value = adapter.value.publicKey;
-        connected.value = adapter.value.connected;
+        if (!wallet?.value) return;
+        publicKey.value = wallet.value.publicKey;
+        connected.value = wallet.value.connected;
     };
     const invalidateListeners = watchEffect((onInvalidate) => {
-        const _adapter = adapter.value;
-        if (!_adapter) return;
+        const _wallet = wallet.value;
+        if (!_wallet) return;
 
-        _adapter.on('connect', onConnect);
-        _adapter.on('disconnect', onDisconnect);
-        _adapter.on('error', onError);
+        _wallet.on('connect', onConnect);
+        _wallet.on('disconnect', onDisconnect);
+        _wallet.on('error', onError);
 
         onInvalidate(() => {
-            _adapter.off('connect', onConnect);
-            _adapter.off('disconnect', onDisconnect);
-            _adapter.off('error', onError);
+            _wallet.off('connect', onConnect);
+            _wallet.off('disconnect', onDisconnect);
+            _wallet.off('error', onError);
         });
     });
 
     if (typeof window !== 'undefined') {
-        // Ensure the adapter listeners are invalidated before refreshing the page.
+        // Ensure the wallet listeners are invalidated before refreshing the page.
```

Comment:
Hmm this looks like a find/replace was done on adapter/wallet which doesn't apply in all cases.

Reply:
True, I'll go back and replace back with adapters where necessary

Reply:
I tried to make the comments more sensible. Let me know if there's anything I should change in particular